redis/src/main/scala/zio/redis/Redis.scala

```diff
@@ -31,14 +30,11 @@ trait Redis
     with api.SortedSets
     with api.Streams
     with api.Scripting
-    with api.Cluster {
-  def codec: BinaryCodec
-  def executor: RedisExecutor
-}
+    with api.Cluster
 
-final case class RedisLive(codec: BinaryCodec, executor: RedisExecutor) extends Redis
+object Redis extends Redis
 
 object RedisLive {
-  lazy val layer: URLayer[RedisExecutor with BinaryCodec, Redis] =
-    ZLayer.fromFunction(RedisLive.apply _)
+  lazy val layer: ULayer[Redis] =
+    ZLayer.succeed(Redis)
```

Comment:
Well, you may as well remove the `RedisLive` object and pack everything in `Redis`, e.g.:

```scala
object Redis {
  lazy val layer: ULayer[Redis] = ZLayer.succeed(new Redis)
}
```

In the future we may consider renaming it to `live`, perhaps it'll be more consistent with the ecosystem.